authors/apps/articles/views.py

```diff
@@ -139,3 +137,43 @@ def put(self, request, pk):
                 {
                     'error': 'You cannot edit an article you do not own. '
                 }, status.HTTP_403_FORBIDDEN)
+
+
+class FavoriteArticle(generics.ListCreateAPIView):
+    """
+    A user is able to favourite an article if had not favourited it.
+    If they had favourited it, thearticle becomes unfavourited.
+    """
+    permission_classes = (IsAuthenticated, )
+    queryset = Article.objects.all()
+    serializer_class = ArticleSerializer
+
+    def post(self, request, slug):
+        """
+        Checks whether the article exists.
```

Comment:
Doc string should start by stating the overview of the method before going deep to the action steps